### PR TITLE
uncomment assertion

### PR DESCRIPTION
**Pull request by uivlis**

### CODE REVIEW

**Comment by tscheinecker** [comment]
you are stilling ignoring my comment from #18 

> in general i don't like that the interface changes
this is a breaking change and makes it potentially hard to upgrade

also you introduce a much harder to use interface
just look at the changes you had to make to the tests, it was incredibly easy to use before, now it isn't

i get that you have to provide a more complex interface for adding the functionality you try to include, but don't do this at the cost of the current interface

-> the `addValue(String text)` method has to stay!

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/PdfReportBuilder.java [hunk 1]
```diff
@@ -22,9 +23,13 @@
 
     PdfReportBuilder addElement(ReportElement element);
 
-    PdfReportBuilder withHeaderOnAllPages(String left, String middle, String right);
+    PdfReportBuilder withHeaderOnAllPages(String left, String middle, String right, boolean excludeOnAllPages);
```
Comment (tscheinecker) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/PdfReportBuilder.java]: here you call the new boolean parameter `excludeOnAllPages` in the impl `excludeOnFirstPage` - which one is it now?

also, once again, this is an interface change. and again the usage is not intuitive, the method is called `OnAllPages` but the last parameter builds an exception to that so it wouldb e `OnAllPagesButTheFirst` ?
would you like to use such an API?

please always use names that actually tell you what is going to happen. also avoid boolean switches.
follow guidelines like the `Clean Code` book by Robert Cecil Martin

also it isn't extensible any more - imagine in your next update you want to allow to add headers only for odd pages. how would you accomplish that?

i would very much like something like this:

- leave the current signature like it is, it doesn't make sense to remove it, just provide an overloaded function and call it with the appropriate defaults (if necessary)
- provide either new functions (appropriately named) like `withHeaderOnAllPagesButFirst` (yes names are allowed to be long!) 
- or provide a more generic api like `withHeaderOnPagesStartingFrom` which accepts a `int startPage` (you could also provide stuff like `withHeaderOnPagesInRange` to have and endPage as well, etc.)
- or another generic approach like `withHeaderOnPages` which accepts an enum with certain values eg. `ALL`, `ALL_BUT_FIRST`, 'ALL_BUT_LAST`, 'EVEN', 'ODD', etc. - this list is exensible, if not dynamic
- or the most generic (and therefor rather cumbersome to use) approach:  `withHeaderOnPages` which accepts a List/Collection/Set/Iterable of page numbers

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/PdfReportBuilder.java [hunk 2]
```diff
@@ -22,9 +23,13 @@
...
 
-    PdfReportBuilder withFooterOnAllPages(String left, String middle, String right);
+    PdfReportBuilder withHeaderOnAllPages(ReportElement headerElement, boolean excludeOnFirstPage);
+
+    PdfReportBuilder withFooterOnAllPages(String left, String middle, String right, boolean excludeOnAllPages);
```
Comment (tscheinecker) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/PdfReportBuilder.java]: here you call the new boolean parameter `excludeOnAllPages` in the impl `excludeOnFirstPage` - which one is it now?

File: cat-boot-report-pdf/src/test/java/cc/catalysts/boot/report/pdf/impl/DemoTest.java [hunk 3]
```diff
@@ -37,8 +50,9 @@ public void demo() throws Exception {
                 .addPadding(10)
                 .startTable()
                 .addColumn("COL1", 2).addColumn("COL2", 2).addColumn("COL3", 4)
-                .createRow().withValues("x1", "x2", "x3")
-                .createRow().withValues("y1", "y2", "y3")
+                .createRow().withValues(new ReportTextBox(styleSheet.getBodyText(), styleSheet.getLineDistance(), "x1"), new ReportImage(img, img.getWidth(), img.getHeight()), new ReportTextBox(styleSheet.getBodyText(), styleSheet.getLineDistance(), "x3"))
+                .createRow().withValues(new ReportTextBox(styleSheet.getBodyText(), styleSheet.getLineDistance(), "y1"), new ReportTextBox(styleSheet.getBodyText(), styleSheet.getLineDistance(), "y3"), new ReportTableBuilderImpl(styleSheet, null)
+                        .addColumn("1", 1).addColumn("2", 3).createRow().withValues("z1", "z2").build())
```
Comment (tscheinecker) [inline cat-boot-report-pdf/src/test/java/cc/catalysts/boot/report/pdf/impl/DemoTest.java]: please revert back to the old string based test and write a new one for the new api

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/PdfReportBuilder.java (hunk 2 above)
Comment (uivlis) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/PdfReportBuilder.java]: auto-completion typo, will fix

**Pull request merged**